src/test/kotlin/io/icure/md/client/apis/e2e/UserApiImplTest.kt

```diff
@@ -1,50 +1,107 @@
+//    private val healthcareProfessionalPublicKey =
+//        runBlocking { TestUtils.healthcareProfessionalPublicKey(iCurePath, authHeader, healthcareProfessionalId) }
 
-    private val medTechApi = MedTechApi.Builder()
-        .iCureUrlPath(iCurePath)
-        .authorization(authHeader)
-        .addKeyPair(healthcareProfessionalId, healthcareProfessionalPublicKey, healthcareProfessionalPrivateKey)
-        .build()
+//    private val medTechApi = MedTechApi.Builder()
+//        .iCureUrlPath(iCurePath)
+//        .authorization(authHeader)
+//        .addKeyPair(healthcareProfessionalId, healthcareProfessionalPublicKey, healthcareProfessionalPrivateKey)
+//        .build()
 
-    private val testedInstance: UserApi = UserApiImpl(medTechApi)
+    // private val anonymousMedTechApi = AnonymousMedTechApi.Builder().authProcessId(authProcessId = "f0ced6c6-d7cb-4f78-841e-2674ad09621e").iCureUrlPath("http://127.0.0.1:16043").build()
+    private val anonymousMedTechApi =
+        AnonymousMedTechApi.Builder().authProcessId(authProcessId = "6a355458dbfa392cb56244031907f47a")
+            .iCureUrlPath("http://127.0.0.1:16043").build()
 
     @Test
-    fun createOrModifyUser_Success_Creation() {
+    fun selfCreatedUser() {
         runBlocking {
-            // When
-            val createdUser =
-                testedInstance.createOrModifyUser(User(login = "hello-${Instant.now().toEpochMilli()}"))
+            val emailAddress = MailUtils.getEmailAddress()
+            val userName = UUID.randomUUID().toString().take(8)
+            val process = anonymousMedTechApi.authenticationApi.startAuthentication(
+                "171f186a-7a2a-40f0-b842-b486428c771b",
+                userName,
+                "",
+                emailAddress,
+                "a58afe0e-02dc-431b-8155-0351140099e4"
+            )
+
+            println("UserName: $userName")
+
+            assert(process != null)
+            println("ProcessId: ${process!!.processId}")
+            println("Login: ${process.login}")
+
+            delay(10000)
+            val email = MailUtils.readInbox(emailAddress).firstOrNull()
+            assert(email != null)
+
+            val validationCode = email!!.mail_subject.takeLast(6)
+            val keyPair = CryptoUtils.generateKeyPairRSA()
+            val tokenAndKeyProvider: ((String, String) -> Triple<String, String, String>?) =
+                { _: String, _: String ->
+                    null
+                }
+            val result = anonymousMedTechApi.authenticationApi.completeAuthentication(
+                process,
+                validationCode,
+                keyPair,
+                tokenAndKeyProvider
+            )
+
+            val api = result.medTechApi
+            val currentUser = api.userApi().getLoggedUser()
+
+            currentUser.patientId?.let { patientId ->
+                val patient = api.patientApi().getPatient(patientId)
+                assert(patient != null)
+            }
```

Comment:
Add assertions to check if : 
- user.patientId != null;
- patient.publicKey == keyPair.publicKey;
- patient.delegations contains patient.id;
- patient.encryptionKeys contains patient.id;
- patient.hcPartyKeys contains patient.id;